ml-agents/mlagents/utils/push_to_hf.py

```diff
@@ -0,0 +1,182 @@
+import os
+import argparse
+
+import yaml
+import json
+
+from pathlib import Path
+
+from huggingface_hub import HfApi, create_repo, upload_folder
+from huggingface_hub.repocard import metadata_save
+
+from typing import Optional
+
+def _generate_config(repo_local_path, configfile_name):
+    """
+    Generate a config.json file from configuration.yaml
+    To do that we convert yaml to json
+    :param repo_local_path: path of the local repository
+    :param configfile_name: name of the yaml config file (by default configuration.yaml)
+    """
+    # Read the YAML file and generate a config.json
+    with open(os.path.join(repo_local_path, configfile_name), "r") as yaml_in, open(
+        os.path.join(repo_local_path, "config.json"), "w") as json_out:
+        yaml_object = yaml.safe_load(yaml_in)
+        json.dump(yaml_object, json_out)
+
+
+def _generate_metadata(model_name: str, env_id: str):
+    """
+    Define the tags for the model card
+    :param model_name: name of the model
+    :param env_id: name of the environment
+    """
+    env_tag = "ML-Agents-" + env_id
+
+    metadata = {}
+    metadata["library_name"] = "ml-agents"
+    metadata["tags"] = [
+        env_id,
+        "deep-reinforcement-learning",
+        "reinforcement-learning",
+        env_tag,
+    ]
+
+    return metadata
+
+
+def _generate_model_card(repo_local_path, configfile_name, repo_id):
```

Comment:
Sometimes you have typings in params, sometimes you don't. I would aim for consistency here :)

Reply:
Updated ✅